### PR TITLE
feat(DescriptionList): add responsive layout breakpoint property

### LINKED ISSUE #5445: Description list: responsive horizontal and vertical layout support
https://github.com/patternfly/patternfly/pull/3815/

Allows you to switch between vertical and horizontal at different breakpoints via:

* `.pf-m-horizontal{-on-[sm, md, lg, xl, 2xl]}`
* `.pf-m-vertical{-on-[sm, md, lg, xl, 2xl]}`

Also we added the small breakpoint to responsive modifiers, so we can add the small breakpoint to:

* `autoFitMinModifier`
* `columnModifier`

### PR DESCRIPTION
**Pull request by kmcfaul**
**What**: Closes #5445

- Adds `layouts` property to specify layout orientation per breakpoint
- Adds `sm` breakpoint to `columnModifier` and `autoFitMinModifier`
- Fixes example code using `autoFitModifier` (instead of `autoFitMinModifier`)

### CODE REVIEW

File: packages/react-core/src/components/DescriptionList/DescriptionList.tsx [hunk 1]
```diff
@@ -27,13 +27,23 @@ export interface DescriptionListProps extends Omit<React.HTMLProps<HTMLDListElem
...
   columnModifier?: {
     default?: '1Col' | '2Col' | '3Col';
+    sm?: '1Col' | '2Col' | '3Col';
     md?: '1Col' | '2Col' | '3Col';
     lg?: '1Col' | '2Col' | '3Col';
     xl?: '1Col' | '2Col' | '3Col';
     '2xl'?: '1Col' | '2Col' | '3Col';
   };
+  /** Indicates how the menu will align at screen size breakpoints. Default alignment is set via the position property. */
+  layouts?: {
```
Comment (mcoker) [inline packages/react-core/src/components/DescriptionList/DescriptionList.tsx:37]: We use `orientation` in the sidebar component with a value of stack (vertical) and split (horizontal). Do you think this prop should match? I'm wondering if we should update the values to be vertical/horizontal to match, too. WDYT?

**Review by mcoker — COMMENTED** [review]
Looks great!! Just one question about the prop/value names.

File: packages/react-core/src/components/DescriptionList/DescriptionList.tsx (hunk 1 above)
Comment (kmcfaul) [reply packages/react-core/src/components/DescriptionList/DescriptionList.tsx:37]: Yeah that sounds good to me

Comment (kmcfaul) [reply packages/react-core/src/components/DescriptionList/DescriptionList.tsx]: Updated the prop name and value order.

**Review by mcoker — APPROVED** [review]
🥳

**Review by tlabaj — APPROVED** [review]

**Review by redallen — APPROVED** [review]

**Review by dlabrecq — COMMENTED** [review]
LGTM

**Pull request merged**